src/main/kotlin/org/rust/lang/core/psi/util/RustPsiExtensions.kt

```diff
@@ -38,8 +43,28 @@ val PsiElement.parentRelativeRange: TextRange?
         TextRange(startOffsetInParent, startOffsetInParent + textLength)
     }
 
+/**
+ * Utility checking whether this particular element precedes the other one
+ */
 fun PsiElement.isBefore(other: PsiElement): Boolean = textOffset < other.textOffset
 
+/**
+ * Utility checking whether this particular element precedes text-anchor (offset)
+ */
+fun PsiElement.isBefore(anchor: Int): Boolean = textOffset < anchor
+
+
+fun PsiElement.getCrate(): Module =
+    ModuleUtilCore.findModuleForPsiElement(this)!!
+
+
+val PsiFile.modulePath: RustModulePath?
```

Comment:
Am I correct that this thing ultimately includes a relative path from the crate root to this file? Looks like the key in index still depends on the names of the parent directories.